scripts/add_headers.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# Copyright 2023 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if ! hash addlicense 2>/dev/null; then
+  go install github.com/google/addlicense@latest
+fi
+
+# adds a new line between the header and the package keyword when it does not exists.
+# we use sed to make in-file text replacements, but sed works differently depending on the version
+if ! sed -i '1s/^/test/' $(mktemp) 2> /dev/null; then
+    # macOS (BSD) version of sed
+    alias gsed="sed -i ''"
+else
+    # POSIX compliant version of sed
+    alias gsed="sed -i"
+fi
+export gsed
+
+find . -type f -name "*.go" -exec gsed -e '/\/\/ limitations under the License\.$/{' -e 'N' -e 's/\(\/\/ limitations under the License.\)\n\(package\)/\1\n\n\2/;}' {} \;
+
+# add license to new files that does not contains it
+shopt -s globstar
+"${GOPATH}"/bin/addlicense  -v -c "IBM Corporation" -l=apache Dockerfile ./**/*.go scripts
```

Comment:
hmm this one failed for me on Linux. Maybe keep your original version, but add a error message with a clue to what could be wrong?

```Bash

if ! hash addlicense 2>/dev/null; then
  go install github.com/google/addlicense@latest
fi

# make sure we can actually use the newly install addlicense script
if ! command -v addlicense &> /dev/null; then
  echo "Can not find 'addlicense' after running 'go install github.com/google/addlicense'."
  echo 'Did you forget to export PATH=$PATH:$GOPATH/bin'
  exit 1
fi

...

# add license to new files that does not contains it
shopt -s globstar
addlicense  -v -c "IBM Corporation" -l=apache Dockerfile **/*.go scripts

```

Reply:
yeah, it make sense :)

Reply:
Ah, my earlier review comment might have been obsolete (sorry). I had not noticed you also changed the Dockerfile and after I noticed that change, I forgot to rebuild my local developer image.

So, to get everything consistent now, should we remove the `"${GOPATH}"/bin/` on the last line since the script would have exited above already.

Reply:
Yes, indeed.